backend/src/main/java/com/engineering/thesis/backend/config/security/SecurityConfig.java

```diff
@@ -27,6 +29,7 @@
 @Configuration
 @RequiredArgsConstructor
 @EnableWebSecurity
+@Order(Ordered.HIGHEST_PRECEDENCE)
```

Comment:
it's required due to org.zalando dependency. Without it spring security wouldn't work properly.